src/main/java/com/desiato/puresynth/controllers/UserController.java

```diff
@@ -1,29 +1,48 @@
 package com.desiato.puresynth.controllers;
 
+import com.desiato.puresynth.exceptions.InvalidTokenException;
+import com.desiato.puresynth.models.CustomUserDetails;
 import com.desiato.puresynth.models.User;
+import com.desiato.puresynth.models.UserDTO;
+import com.desiato.puresynth.services.AuthenticationService;
 import com.desiato.puresynth.services.UserService;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
-import org.springframework.security.core.Authentication;
-import org.springframework.security.core.GrantedAuthority;
-import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.web.bind.annotation.*;
 
-import java.util.Optional;
-
 @RestController
 @RequestMapping("/api/user")
 public class UserController {
 
-    @Autowired
-    private UserService userService;
+    private final UserService userService;
+    private final PasswordEncoder passwordEncoder;
+    private final AuthenticationService authenticationService;
+
+    public UserController(UserService userService, PasswordEncoder passwordEncoder, AuthenticationService authService) {
+        this.userService = userService;
+        this.passwordEncoder = passwordEncoder;
+        this.authenticationService = authService;
+    }
 
-    @GetMapping
-    public String helloApi(){
-        return "hello api";
+    private static final Logger logger = LoggerFactory.getLogger(UserController.class);
+
+    @GetMapping("/me")
+    public ResponseEntity<?> getProtectedEndpoint(@RequestHeader("authToken") String token) throws InvalidTokenException {
+        UserDetails userDetails = authenticationService.loadUserByToken(token);
+        if (userDetails == null) {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body("Unauthorized: Invalid or missing token");
+        } else if (!(userDetails instanceof CustomUserDetails)) {
```

Comment:
Why this cast? Why not make authenticationService.loadUserByToken return `CustomUserDetails`?